plugin.video.jafc/resources/lib/plugin.py

```diff
@@ -0,0 +1,300 @@
+# -*- coding: utf-8 -*-
+"""Main plugin file - Handles the various routes"""
+
+__author__ = "fraser"
+
+import routing
+import xbmc
+import xbmcaddon as xa
+import xbmcplugin as xp
+from xbmcgui import ListItem
+
+from resources.lib import kodiutils as ku
+from resources.lib import search as jafc
+
+plugin = routing.Plugin()
+ADDON = xa.Addon()
+ADDON_ID = ADDON.getAddonInfo("id")  # plugin.video.jafc
+ADDON_NAME = ADDON.getAddonInfo("name")  # East Anglian Film Archive
+MEDIA_URI = "special://home/addons/{}/resources/media/".format(ADDON_ID)
```

Comment:
Please use API: https://codedocs.xyz/xbmc/xbmc/group__python__xbmcaddon.html

Reply:
ADDON_PATH  = ADDON_ID.getAddonInfo('path').decode('utf-8')
MEDIA_URI  = os.path.join(ADDON_PATH , 'resources', 'media','')

Reply:
Why on earth would I use os.path to join a special path URL - that makes no sense at all as the special path itself is path translated via xbmc.translatePath

Reply:
Just looking at this again and it really doesn't make any sense. 
Your suggested change...

ADDON_PATH = ADDON_ID.getAddonInfo('path').decode('utf-8')

AttributeError: 'str' object has no attribute 'getAddonInfo'

Reply:
Presume it would be 

ADDON_PATH = ADDON.getAddonInfo('path').decode('utf-8')

...

Reply:
Yes it should be a `xbmcaddon.Addon()` instance